modules/statistics/src/main/frontend/src/Statistics/Statistic.jsx

```diff
@@ -0,0 +1,171 @@
+      field["x"] = DateQuestionUtilities.amendMoment(field["x"], dateFormat).format(moment.HTML5_FMT.DATE);
+      return field;
+    })
+  } else if (["long", "double", "decimal"].includes(definition["xVar"]["dataType"])) {
+    // Numeric sort
+    rechartsData.sort((a, b) => a["x"] - b["x"]);
+  } else {
+    // Rely on the default (string-coerced) sort
+    rechartsData.sort((a, b) => a["x"].localeCompare(b["x"]));
+  }
+
+  let allFields = Object.keys(allFieldsDict);
+  if (definition["splitVar"]) {
+    // Reorder according to splitVar's order
+    allFields = Object.values(definition["splitVar"])
+      // Only grab answer options
+      .filter((field) => field["jcr:primaryType"] == "cards:AnswerOption"
+      // Furthermore, filter it to only fields that exist in the data
+        && field["label"] in allFieldsDict)
+      // Sort according to defaultOrder (if they exist)
+      .sort((option1, option2) => (option1.defaultOrder - option2.defaultOrder))
+      .map((field) => field["label"]);
+    // If there are any fields in our data that aren't in the splitVar's answerOptions, include
+    // them at the end
+    allFields = allFields.concat(
+      Object.keys(allFieldsDict)
+        .filter((field) => !allFields.includes(field))
+      );
+  }
+
+  let isBar = definition["type"] == "bar";
+
+  let icon = isBar ? <BarChartIcon /> : <LineChartIcon />;
+  let iconColor = isBar ? deepPurple[400] : indigo[400];
+
+  let ChartType = isBar ? BarChart : LineChart;
+  // Generate a palette via google-palette, and append the necessary # for a hex value
+  let chartColours = DEFAULT_PALETTE.slice();
+  let numKeys = Object.keys(allFields).length;
+  if (numKeys > DEFAULT_PALETTE.length) {
+    chartColours = chartColours.concat(palette("rainbow", numKeys-DEFAULT_PALETTE.length).map((col) => "#" + col));
+  }
+
+  let yAxisLabel = `${definition["y-label"]}s`;
+  let widgetHeight = 250; // TODO: make mobile friendly
+
+  return <Grid item md={12} lg={6}>
+    <Card>
+      <CardHeader
+        disableTypography
+        avatar={<Avatar style={{background: iconColor }}>{icon}</Avatar>}
+        title={<Typography variant="h6">{definition["name"]}</Typography>}
+        />
+      <CardContent>
+      { allFields.length == 0 ?
+        <Grid container direction="row" justify="center" alignItems="center" style={{height: widgetHeight}}>
+          <Grid item>
+            <Typography color="textSecondary" variant="caption">No data available for this statistic</Typography>
+          </Grid>
+        </Grid>
+        :
+        <ResponsiveContainer width="100%" height={widgetHeight}>
+          <ChartType data={rechartsData}>
+            <CartesianGrid strokeDasharray="3 3" />
+            <XAxis dataKey="x">
+              <Label value={definition["x-label"]} offset={0} position="insideBottom" />
+            </XAxis>
+            <YAxis allowDecimals={false} label={{ value: yAxisLabel, angle: -90, position: 'insideLeft' }} />
```

Comment:
We could probably also bring the Y label closer to the Y axis to make everything more balanced:
```suggestion
            <YAxis allowDecimals={false} label={{ value: yAxisLabel, angle: -90, position: 'insideLeft', offset: 10 }} />
```